sql/hive-thriftserver/src/main/java/org/apache/hive/service/BreakableService.java

```diff
@@ -77,21 +77,21 @@ private void maybeFail(boolean fail, String action) {
   }
 
   @Override
-  public void init(HiveConf conf) {
```

Comment:
This is overriding the method of following class.
https://github.com/apache/spark/blob/ab048990e0bdffe16de6a6b9211f63933976651e/sql/hive-thriftserver/v2.3/src/main/java/org/apache/hive/service/AbstractService.java#L84-L90

Reply:
Therefore it should also be synchronized.